Add tests to check VCN activity tracing in perfetto output

sputhala-amd opened the pull request:
For testing the output of VCN activity a sample from the rocDecode repo is used: https://github.com/ROCm/rocDecode/tree/develop/samples/videoDecodeBatch

The tests currently is ensuring that the perfetto output captures VCN activity in the trace.

sputhala-amd committed:
Add tests to capture VCN tracing

For testing the output of VCN activity a sample from the rocDecode repo is used:
https://github.com/ROCm/rocDecode/tree/develop/samples/videoDecodeBatch

The tests currently is ensuring that the perfetto output captures VCN
activity in the trace.

dgaliffiAMD committed:
Add rocdecode-dev and others to workflow setup

- DEB - rocdecode-dev libavformat-dev libavcodec-dev
- RPM- rocdecode-devel libavcodec-free-devel libavformat-free-devel

dgaliffiAMD committed:
Fix to cmake file

sputhala-amd committed:
Install rocdecode base

dgaliffiAMD commented on external/dyninst:
I don't think you meant to update the `dyninst` submodule here.

sputhala-amd replied:
Yes this was an error something to so with my workspace automatically doing this somehow.. I have reverted the changes

dgaliffiAMD commented on examples/videodecode/CMakeLists.txt:
```diff
@@ -0,0 +1,144 @@
+
+# Find FFMPEG 
+find_package(FFmpeg QUIET)
+if(NOT FFmpeg_FOUND)
```
```suggestion
if(NOT FFMPEG_FOUND)
```

dgaliffiAMD committed:
Update workflows.

- Exclude the "videodecode" test, since there are no GPUs.
- Install DEBs for rocdecode-dev libavformat-dev libavcodec-dev
- Install RPMs for rocdecode-devel libavcodec-free-devel libavformat-free-devel
- Add /etc/apt/sources.list.d/amdgpu.list because some packages are required by rocdecode.

dgaliffiAMD committed:
Workflow updates

dgaliffiAMD committed:
Update workflows

dgaliffiAMD committed:
RHEL workflow update

Skip install of ffmpeg for RHEL 8.

dgaliffiAMD committed:
Update videodecode example to align with rocm-6.3.1 headers

sputhala-amd committed:
Additional changes to RedHat build

sputhala-amd committed:
Revert the changes made to dyninst done by error

dgaliffiAMD commented on cmake/ConfigCPack.cmake:
```diff
@@ -171,6 +171,11 @@ if(ROCPROFSYS_USE_MPI)
         list(APPEND _DEBIAN_PACKAGE_DEPENDS "libmpich-dev")
     endif()
 endif()
+if(ROCPROFSYS_BUILD_TESTING)
+    list(APPEND _DEBIAN_PACKAGE_DEPENDS "rocdecode")
+    list(APPEND _DEBIAN_PACKAGE_DEPENDS "rocdecode-dev")
+    list(APPEND _DEBIAN_PACKAGE_DEPENDS "rocdecode-test")
+endif()
```
```suggestion
if(ROCPROFSYS_BUILD_TESTING)
    list(APPEND _DEBIAN_PACKAGE_DEPENDS "rocdecode-test")
endif()
```
You should be able to just include "rocdecode-test", this package will take care of the other dependencies.

dgaliffiAMD commented on examples/videodecode/common.h:
```diff
@@ -0,0 +1,102 @@
+/*
+Copyright (c) 2023 - 2024 Advanced Micro Devices, Inc. All rights reserved.
```
```suggestion
Copyright (c) 2023 - 2025 Advanced Micro Devices, Inc. All rights reserved.
```

dgaliffiAMD commented on examples/videodecode/roc_video_dec.h:
```diff
@@ -0,0 +1,650 @@
+
+/*
+Copyright (c) 2023 - 2024 Advanced Micro Devices, Inc. All rights reserved.
```
```suggestion
Copyright (c) 2023 - 2025 Advanced Micro Devices, Inc. All rights reserved.
```

dgaliffiAMD commented on examples/videodecode/video_demuxer.h:
```diff
@@ -0,0 +1,716 @@
+/*
+Copyright (c) 2023 - 2024 Advanced Micro Devices, Inc. All rights reserved.
```
```suggestion
Copyright (c) 2023 - 2025 Advanced Micro Devices, Inc. All rights reserved.
```

dgaliffiAMD commented on examples/videodecode/videodecodebatch.cpp:
```diff
@@ -0,0 +1,707 @@
+#include <atomic>
```
```suggestion
/*
Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.
*/

#include <atomic>
```

dgaliffiAMD commented on examples/videodecode/roc_video_dec.cpp:
```diff
@@ -0,0 +1,1605 @@
+/*
+Copyright (c) 2023 - 2024 Advanced Micro Devices, Inc. All rights reserved.
```
```suggestion
Copyright (c) 2023 - 2025 Advanced Micro Devices, Inc. All rights reserved.

```

dgaliffiAMD reviewed (changes requested):
Updating the dates in the source code headers now that it is a new year.

dgaliffiAMD commented on tests/rocprof-sys-videodecode-tests.cmake:
```diff
@@ -0,0 +1,34 @@
+        --print-instrumented
+        functions
+        --print-instructions
+    LABELS "videodecode")
```
```suggestion
rocprofiler_systems_add_test(
    SKIP_BASELINE SKIP_RUNTIME SKIP_REWRITE
    NAME videodecode
    TARGET videodecode
    GPU ON
    RUN_ARGS -i ${ROCmVersion_DIR}/share/rocdecode/video/ -t 2
    LABELS "videodecode")
```
Added `GPU ON`, so test will check for a valid GPU before running. 
Also, consider simplifying the test for this iteration, since we're just testing for the VCN Activity track. For the moment, we can add `SKIP_RUNTIME` to save on testing time.

dgaliffiAMD commented on tests/rocprof-sys-videodecode-tests.cmake:
```diff
@@ -0,0 +1,34 @@
+    LABELS "videodecode")
+
+rocprofiler_systems_add_validation_test(
+    NAME videodecode-perfetto-sampling
```
```suggestion
    NAME videodecode-sampling
```
Fixing the name that the validation test depends on.

dgaliffiAMD commented on tests/rocprof-sys-videodecode-tests.cmake:
```diff
@@ -0,0 +1,34 @@
+
+rocprofiler_systems_add_validation_test(
+    NAME videodecode-perfetto-sampling
+    PERFETTO_FILE "perfetto-trace.proto"
+    LABELS "videodecode"
+    ARGS --key-names vaInitialize --key-counts 1)
```
```suggestion
    PERFETTO_METRIC "host"
    PERFETTO_FILE "perfetto-trace.proto"
    LABELS "videodecode"
    ARGS -l videodecode -c 1 -d 0)
```
Updated the validation parameters, but this will just be a basic "did it run" type of test. We'll have to modify the validate-perfetto-proto.py (or create a new one) to validate that the VCN counters are present.

sputhala-amd replied:
You are right ... I missed including this change ...

dgaliffiAMD committed:
Updating testing parameters

dgaliffiAMD committed:
Updating testing parameters

sputhala-amd committed:
Merge branch 'sajina/vcn_testing' of https://github.com/ROCm/rocprofiler-systems into sajina/vcn_testing

sputhala-amd committed:
Format the python file

Fix formatting errors.

dgaliffiAMD commented on tests/validate-perfetto-proto.py:
```diff
@@ -175,6 +182,19 @@ def validate_perfetto(data, labels, counts, depths):
+        total_value = 0
+        for row in sum_counter_values:
+            total_value = row.total_value if row.total_value is not None else -1
+        print(total_value, counter_name)
```
```suggestion
```
Removing print statement ?

dgaliffiAMD reviewed (dismissed):
Thanks. It looks good. Maybe just remove that one print statement in the validation script. It looks more like a debug log.

sputhala-amd committed:
Update tests/validate-perfetto-proto.py

Remove the debug print

Co-authored-by: David Galiffi <David.Galiffi@amd.com>
